### PR TITLE
3869 fix default search engine

### LINKED ISSUE #3869: [Bug] Fenix not setting default search engine properly
## Steps to reproduce
Set locale to RU or CN

### Expected behavior
Default search engine is not Google

### Actual behavior
Default search engine is Google

### Device information
any

### PR DESCRIPTION
**Pull request by Baron-Severin**
---

NOTE: this does not close #3869, which is still blocked on getting an API key.  This completes the prework so we will be able to just swap the key in and close.

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
No user facing changes
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features
No user facing changes

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/components/metrics/GleanMetricsService.kt [hunk 1]
```diff
@@ -468,19 +472,21 @@ class GleanMetricsService(private val context: Context) : MetricsService {
...
+                    .searchEngineManager
+                    .getDefaultSearchEngineAsync(context)
 
-            code.set(defaultEngine.identifier)
-            name.set(defaultEngine.name)
-            submissionUrl.set(defaultEngine.buildSearchUrl(""))
-        }
+                code.set(defaultEngine.identifier)
+                name.set(defaultEngine.name)
+                submissionUrl.set(defaultEngine.buildSearchUrl(""))
```
Comment (Baron-Severin) [inline app/src/main/java/org/mozilla/fenix/components/metrics/GleanMetricsService.kt]: As far as I can tell, these properties are only used to send this ping, so making this async shouldn't change any behavior.

File: app/src/main/java/org/mozilla/fenix/search/SearchFragment.kt [hunk 2]
```diff
@@ -297,6 +308,35 @@ class SearchFragment : Fragment(), BackHandler {
         } else null
     }
 
+    /**
+     * Attempts to get a cached search engine. If none is immediately available, it falls back
+     * to a default.
+     *
+     * Cache hits/misses are reported to telemetry.
+     */
+    private fun getCurrentSearchEngine(context: Context): SearchEngine {
```
Comment (Baron-Severin) [inline app/src/main/java/org/mozilla/fenix/search/SearchFragment.kt]: See commit message for an explanation as to why this approach was taken.

File: app/src/main/java/org/mozilla/fenix/search/SearchFragment.kt [hunk 3]
```diff
@@ -284,6 +297,32 @@ class SearchFragment : Fragment(), BackHandler {
...
+            runBlocking { deferredSearchEngine.await() }
+        } else {
+            null
+        }
+
+        val backupSearchEngine = context.components.search.localeSearchEngineManager
+            .getDefaultSearchEngine(context)
+
+        val cacheWasHit = selectedSearchEngine != null
+        Events.searchFragmentCacheAccess.set(cacheWasHit)
```
Comment (Baron-Severin) [inline app/src/main/java/org/mozilla/fenix/search/SearchFragment.kt]: This is the first time I've added a new ping, would you please double check that I'm doing it properly?

**Comment by Baron-Severin** [comment]
@boek data review
# Request for data collection review form

1) What questions will you answer with this data?

- How frequently do users reach the SearchFragment without their location already having been cached?

2) Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements?

- If cache misses are frequent, we will need to find a new solution to the problem. If not, this is probably fine.

3) What alternative methods did you consider to answer these questions? Why were they not sufficient?

- This is the only way to determine this information.

4) Can current instrumentation answer these questions?

- No telemetry currently exists to answer these questions.

5) List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox [data collection categories](https://wiki.mozilla.org/Firefox/Data_Collection) on the Mozilla wiki.   

| Measurement Description                      | Data Collection Category | Tracking Bug #                                               |
| -------------------------------------------- | ------------------------ | ------------------------------------------------------------ |
| When `SearchFragment` has been opened, whether or not our search engine lookup hits a warm cache | 1                        | [Issue 3869](https://github.com/mozilla-mobile/fenix/issues/3869 |

6) How long will this data be collected?

* Fenix engineering will monitor this data until 2020-03-01

7) What populations will you measure?

* All users with telemetry enabled, on release channel.


8) If this data collection is default on, what is the opt-out mechanism for users?

- There is an option in settings to stop sending usage data.

9) Please provide a general description of how you will analyze this data.

- The data will be analyzed through Redash queries and dashboards.

10) Where do you intend to share the results of your analysis?

- On Redash and with mobile teams.

File: app/metrics.yaml [hunk 4]
```diff
@@ -154,6 +154,16 @@ events:
     notification_emails:
       - fenix-core@mozilla.com
     expires: "2020-03-01"
+  search_fragment_cache_access:
```
Comment (boek) [inline app/metrics.yaml]: Could you fill out the (https://github.com/mozilla/data-review/blob/master/request.md)[data request form] for this? :)

Comment (Baron-Severin) [reply app/metrics.yaml]: @boek [already did](https://github.com/mozilla-mobile/fenix/pull/6197#issuecomment-547030238)!

Comment (boek) [reply app/metrics.yaml]: I'm a dummy, don't know how I missed it!

**Comment by Baron-Severin** [comment]
Closing, as these changes will be made redundant by #5577.  We expect that story to fix #3869

**Pull request closed without merging**